src/memory_server.ml

```diff
@@ -45,9 +45,9 @@ let wrap dbg f =
 		(* NB both needed and free have been inflated by the lowmem_emergency_pool etc *)
 		let needed = Int64.sub needed Squeeze_xen.target_host_free_mem_kib
 		and free = Int64.sub free Squeeze_xen.target_host_free_mem_kib in
-		raise (Memory_interface.Cannot_free_this_much_memory (needed, free))
+		raise (MemoryError (Memory_interface.Cannot_free_this_much_memory (needed, free)))
```

Comment:
Not automatically, but you can do it. You use `GenClient` or `GenServer` rather than `GenClientExn` or `GenServerExn`. For example, if you declare your errors like this:
```ocaml
type errors =
| MyError1 [@doc ["Error1 doc"]]
| MyError2 [@doc ["Error2 doc"]]
[@@deriving rpcty]

exception ErrorExn of errors

let err = Error.{
  def = errors;
  raiser = (function | e -> raise (ErrorExn e));
  matcher = function | ErrorExn e -> Some e | _ -> None
}
```
and your method like this:

```ocaml
let method = declare "test" ["A test method"] (int @-> string @-> returning int err)
```

then when you use `GenServerExn`, the expected signature of the implementation will be

```ocaml
test: int -> string -> int
```
and it will correctly marshal the error if you `raise ErrorExn` in the middle of your implementation.

However, if you use `GenServer`, the expected type of the implementation will be
```ocaml
test: int -> string -> (int, errors) Result.result
```